Automattic/jetpack
# Jetpack Boost: Add Error Boundaries

## Proposed changes:
- Added `ErrorBoundary` component to wrap `SpeedScore` and `Module` components for better error handling.
- Implemented fallback UI for `SpeedScore` and `Module` when an error occurs.
- Created a new file for the `ErrorBoundary` component to centralize error handling logic.

Here's what a Broken Dashboard looks like now:
![2024_02_08uuTt4SpH 42](https://github.com/Automattic/jetpack/assets/988095/0e659de4-e247-42a9-9b09-3d034c6cda7b)



### Other information:



## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
N/A

## Testing instructions:
- Navigate to the Speed Score feature in Jetpack Boost and ensure it loads correctly.
- `throw` an error in the `SpeedScore` component and check that the fallback UI is displayed.
- Navigate to different modules in Jetpack Boost to verify they load without issues.
-`throw` an error in the `Module` component and observe that the error UI is shown instead of the module.
- Check the console for error logging to ensure `ErrorBoundary` is catching and logging errors as expected.
